### ISSUE TITLE
Associations not working

### ISSUE THREAD

**Issue by goalaleo** [contentful/contentful_model#40]
Here are the classes:
```
class BlogPost < ContentfulModel::Base
  self.content_type_id = "blogPost"
  has_one :author
end

class Author < ContentfulModel::Base
  self.content_type_id = "author"
  belongs_to_many :createdBlogPosts, class_name: "BlogPost"
end
```

trying to use associations just returns `nil`for `has_one`, and an empty array for `has_many`:

```
post = BlogPost.first
 #=> returns the blogpost with the actual :author in fields
post.author
#=> nil

author = Author.first
#=> returns the author of the blogpost, :createdBlogPosts includes the above post
author.createdBlogPosts
#=> []
```
They don't throw `ContentfulModel::AttributeNotFoundError`, so the attribute exists is, but nothing gets returned. Any advice?

**Comment by goalaleo** [comment]
I got the associations to work, but the solution doesn't make any sense to me.
It was to remove the `has_one :author`association from the `BlogPost` class.
Can you come up with a reason why the existence of the association would screw things up?

**Comment by dlitvakb** [comment]
Hey @goalaleo,

I'll be looking into it

Cheers

**Comment by pmoore** [comment]
@dlitvakb @goalaleo I think I'm running into the same issues now, and think it may be from the commit here: https://github.com/contentful/contentful_model/commit/35053665a74e7d8512d2ecdad3fee7839d6b07ac#diff-4a8a1bd3b58710c02288eee36bb77fdbR10

Looks like the initializer is now calling `defined_getters` for all the fields that are returned, which override the methods defined by the `has_many`, `has_one`, etc associations.   I'm not sure why this is happening, but pretty much renders the association helper methods useless as their code will never be touched, unless I'm missing somehting.

**Comment by dlitvakb** [comment]
Hey @pmoore,

Looks like a possible cause. 

I'll investigate more on it and how to avoid creating a generic getter for association methods.

Thanks for the input.

**Comment by pmoore** [comment]
not pretty, but as a workaround i'm overriding the `define_getters` method to not actually do anything and everything else still seems to work...so far

```
class Post < ContentfulModel::Base
   
    private

    def define_getters
      # Overriding method for now
    end
end
```

**Comment by jpowell** [comment]
Could this be a product of not having the correct `load_depth` specified for your model?

**Comment by nomasprime** [comment]
@jpowell what is `load_depth`? Searching repo doesn't reveal anything.

**Comment by jpowell** [comment]
@nomasprime good call. That's an alias for `load_children` I've defined in a mixin supporting some defaults for models. Updated comment above. Thanks!

**Comment by nomasprime** [comment]
Thanks @jpowell. This solves my immediate problem.

Would have thought this could have been done dynamically though unless I'm missing something, i.e., why can't `model.model.model` automatically set `load_children` to `2`?

**Comment by jpowell** [comment]
The implementation is not a lazy evaluation, like an `ActiveRecord::Relation`, per say, but the call is being evaluated immediately... they don't load dependencies / children by default because of how expensive it could be, but provide you the back door so you can shoot yourself in the foot if you want ;)

From what I can tell, defining associations really does nothing. You don't have to define them to be able to call `model.child`. When I found this gem, I was hoping calling `model.child` when i have `has_one :child` defined would actually return an instance of my domain model that inherits from `ContentfulModel::Base`, but alas. So, you'd have to build a wrapper class for your `ContentfulModel`s and manually instantiate it.

**Comment by dlitvakb** [comment]
Hey @jpowell,

The API by default has `include: 1`, which is equivalent to `load_children(1)`, the maximum value is 10.

Cheers
